File: Old_xml_and_scripts/Jim/plan/old-predUcesanim/PlanSch.rb
########	plan.rb	 ########
#	@author: Team17		#
#############################

class PlanSch < Plan
  
  def initialize
    super
  end
  
  def replan
    
    @plan.clear
#    if !(EnvironmentModel.beamablePlayMode)
      @agentInfo.whereIsBall();
      @agentInfo.isBallMine();
      @agentInfo.nearBall();
      @agentInfo.ballControlPosition();
  	
      @target_position = @agentInfo.ballControlPosition()
      @target_position_phi = @target_position.getPhi()
      @kick_target = @agentInfo.kickTarget()
      @kick_target_phi =  @kick_target.getPhi()
      
      
      if (me.on_ground? or me.is_lying_on_back? or me.is_lying_on_belly?)
        @agentInfo.loguj('GetUp');
        @plan << GetUp.new(Proc.new{me.on_ground?})
      elsif (not see_ball?) 
        puts "Searching ball"
        @agentInfo.loguj('Localize');
        @plan << Localize.new(Proc.new{(not see_ball?)})
      else
        @targetPosition = Java::sk.fiit.robocup.library.geometry.Vector3D.cartesian(5, 5, 0)
        @agentInfo.loguj('GoToBall');
       	@plan << GotoBall.new(@targetPosition)
      end
#   	end
  end
  
end

File: Jim/src/sk/fiit/jim/garbage/plan/PlanSch.java
package sk.fiit.jim.garbage.plan;

import sk.fiit.jim.agent.AgentInfo;
import sk.fiit.jim.agent.highskill.*;
import sk.fiit.jim.agent.models.AgentModel;
import sk.fiit.jim.agent.models.WorldModel;
import sk.fiit.robocup.library.geometry.Vector3D;

/**
 * Rewrited PlanSch from ruby
 * 
 * @author Bosiak
 */

public class PlanSch extends Plan {
	public static PlanSch getInstance() {
		return new PlanSch();
	}

	public void replan() {
		AgentModel agentModel = AgentModel.getInstance();
		Vector3D target_position = AgentInfo.getInstance()
				.ballControlPosition();
		if (agentModel.isOnGround() || agentModel.isLyingOnBack()
				|| agentModel.isLyingOnBelly()) {

			AgentInfo.logState("GetUp");
			Highskill_Queue.add(new GetUp());
		} else if (!WorldModel.getInstance().isSeeBall()) {
			AgentInfo.logState("Localize");
			Highskill_Queue.add(new Localize());
		} else {
			AgentInfo.logState("GoToBall");
			Highskill_Queue.add(new GotoBall(target_position));

		}
	}
}
